### PR TITLE
Add integration with Edge browser (Chromium-based)

### PR DESCRIPTION
**Pull request by varjolintu**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (non-breaking change which adds functionality)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )
All paths used must be verified after the stable version of Edge is released. This PR is WIP until then.

https://microsoftedge.microsoft.com/addons/detail/pdffhmdngciaglkoonimfcmckehcpafo

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Manually.

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ My change requires a change to the documentation, and I have updated it accordingly.

### CODE REVIEW

**Comment by iamkarlson** [comment]
Any chance to get a binary so people can start testing it already?

**Comment by varjolintu** [comment]
@iamkarlson Not until the final Edge version is released so we can be sure what the final configuration and registry paths are.

**Comment by iamkarlson** [comment]
@varjolintu how do you think is it possible to mark this integration as beta so you can actually roll this out? if any path is changed in the future then you just create non-beta integration with actual paths

**Comment by varjolintu** [comment]
@iamkarlson We don't generally support any Beta or Nightly browser versions, so it's best to wait for the stable. You can already modify/copy the necessary files manually to get the current Edge release to work. Please see this thread for further information: https://github.com/keepassxreboot/keepassxc-browser/issues/46

**Comment by varjolintu** [comment]
Linux version has been confirmed.

**Comment by varjolintu** [comment]
`pdffhmdngciaglkoonimfcmckehcpafo` will be the extension ID when installed from Microsoft Extension Store.

File: src/browser/HostInstaller.cpp [hunk 1]
```diff
@@ -139,8 +142,8 @@ void HostInstaller::installBrowser(SupportedBrowsers browser,
  */
 void HostInstaller::updateBinaryPaths(const bool& proxy, const QString& location)
 {
-    // Where 6 is the number of entries in the SupportedBrowsers enum declared in HostInstaller.h
-    for (int i = 0; i < 6; ++i) {
+    // Where 7 is the number of entries in the SupportedBrowsers enum declared in HostInstaller.h
```
Comment (z3ntu) [inline src/browser/HostInstaller.cpp]: comment is not accurate with the current change

File: src/browser/HostInstaller.h [hunk 2]
```diff
@@ -35,9 +35,12 @@ class HostInstaller : public QObject
         FIREFOX = 2,
         VIVALDI = 3,
         TOR_BROWSER = 4,
-        BRAVE = 5
+        BRAVE = 5,
+        EDGE = 6
     };
 
+    static const int SUPPORTED_BROWSERS_COUNT = 7;
```
Comment (z3ntu) [inline src/browser/HostInstaller.h]: I know of several pieces of software where this number is made the last value of the enum but that's preference I guess (and depending on the code style of this project which I am not familiar with)

Comment (droidmonkey) [reply src/browser/HostInstaller.h]: I agree with this approach.

File: src/browser/HostInstaller.cpp (hunk 1 above)
Comment (droidmonkey) [reply src/browser/HostInstaller.cpp]: Recommend removing this comment entirely.

**Comment by droidmonkey** [comment]
I'm downloading Chromified Edge today to test this out. Pretty excited because I'm probably going to make this my primary browser.

**Comment by varjolintu** [comment]
Windows support seems to have some issues. The registry path is created but the value is not changed, or the .json file created. Haven't debugged that yet. macOS support worked right away. Linux support is still in the sources, but for now the checkbox is disabled.

Also the snap helper script needs to be updated when Microsoft releases the Linux version.

**Comment by varjolintu** [comment]
Tried this again. Windows is working normally for me.

**Comment by varjolintu** [comment]
@droidmonkey Better leave it disabled. There's no guarantee what the final path for the configuration files will be. Originally I also updated the helper script, but let's wait with that too.

**Comment by droidmonkey** [comment]
I fixed the code format issue

**Comment by max-w** [comment]
Does not work for me, neither on macOS nor Win10. KeepassXC 2.5.2 and latest extension from the MS Store.

**Comment by droidmonkey** [comment]
You need to run the release/2.5.3 branch at this point... We have not released this feature yet.

**Pull request merged**